addons/sale_pdf_quote_builder/tests/test_pdf_quote_builder.py

```diff
@@ -256,3 +256,29 @@ def _test_custom_content_kanban_like(self):
             login='admin',
         )
         # Assert documents are selected
+
+    def test_quotation_document_is_removed_on_template_change(self):
+        so_tmpl = self.env['sale.order.template'].create({
+            'name': "test1",
+            'quotation_document_ids': [
+                Command.link(self.header.id),
+            ],
+        })
+        so_tmpl_2 = self.env['sale.order.template'].create({
+            'name': "test1",
+        })
+
+        self.sale_order.write({
+            'sale_order_template_id': so_tmpl.id,
+            'quotation_document_ids': [
+                Command.link(self.header.id),
+            ],
+        })
+        self.assertEqual(self.sale_order.quotation_document_ids, self.header)
+
+        self.sale_order.update({
+            'sale_order_template_id': so_tmpl_2.id,
+        })
+
+        self.sale_order._onchange_sale_order_template_id()
```

Comment:
Better to use `Form` instead of raw-dogging `onchange` methods
```suggestion
        so_form = Form(self.sale_order)
        so_form.sale_order_template_id = so_tmpl_2
        so_form.save()
```